core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -124,10 +124,16 @@ private[spark] class DiskBlockManager(blockManager: BlockManager, conf: SparkCon
     (blockId, getFile(blockId))
   }
 
+  /**
+   * Create local directories for storing block data. These directories are
+   * located inside configured local directories and won't
+   * be deleted on JVM exit when using the external shuffle service.
```

Comment:
Thanks for the clarification.

The reason why we saw the shuffle file not found is because we deleted the parent directory on exit, which included the shuffle directory. Your changes made these two directories siblings instead. That part makes sense.

As for who is responsible for cleaning up the shuffle files when the application exits, let's move that discussion back to the main thread.